lucille-core/src/test/java/com/kmwllc/lucille/stage/ParseDateTest.java

```diff
@@ -52,7 +59,119 @@ public void testGetLegalProperties() throws StageException {
             "source",
             "dest",
             "conditions",
-            "class"),
+            "class",
+            "time_zone_id"),
         stage.getLegalProperties());
   }
+
+  @Test
+  public void testInvalidTimeZoneId() {
+    Throwable exception = assertThrows(StageException.class, () -> factory.get("ParseDateTest/invalidTimeZoneConfig.conf"));
+    Throwable cause = exception.getCause().getCause();
+    assertEquals(IllegalArgumentException.class, cause.getClass());
+    assertEquals("Invalid time zone ID: \"invalid\", must be one of TimeZone.getAvailableIDs()", cause.getMessage());
+  }
+
+  @Test
+  public void testParseDateWithTimeZone() throws StageException {
+
+    // create a test config without a time zone
+    HashMap<String,Object> configValues = new HashMap<>();
+    configValues.put("class", "com.kmwllc.lucille.stage.ParseDate");
+    configValues.put("source", List.of("date"));
+    configValues.put("dest", List.of("output"));
+    configValues.put("format_strs", List.of("yyyy-MM-dd"));
+
+    // create a map of time zones to expected formatted dates
+    Map<String, String> timeZoneToFormattedDate = Map.of(
+        "EST", "2021-02-02T00:00:00Z",
+        "Universal", "2021-02-02T00:00:00Z",
+        "Europe/Rome", "2021-02-01T00:00:00Z",
+        "Australia/Brisbane", "2021-02-01T00:00:00Z"
+    );
+
+    for (Map.Entry<String, String> entry: timeZoneToFormattedDate.entrySet()) {
+
+      // get time zone id from the map
+      String timeZoneId = entry.getKey();
+
+      // add a time zone to the config values map
+      configValues.put("time_zone_id", timeZoneId);
+
+      // create a stage off the config values map
+      Stage stage = factory.get(configValues);
+
+      // create a document with a date field
+      Document doc = Document.create("doc");
+      doc.setField("date", "2021-02-02");
+
+      // parse date and confirm it matches the expected formatted date
+      stage.processDocument(doc);
+      assertEquals("Failed for time_zone_id=" + timeZoneId,
+          entry.getValue(), doc.getString("output"));
+    }
+  }
+
+//  @Test
```

Comment:
we'd like to keep commented-out code to a minimum

if this is here to explain something important, leave it, but please:

- remove the enclosing method (i.e. public void test2()) because that makes it look like an actual test that we disabled for some reason
- add a few comments saying why this is here and what it's showing